source/client-side-encryption/client-side-encryption.rst

```diff
@@ -293,6 +296,59 @@ cluster as indicated by the connecting URI. But the optional
 keyVaultClient can be used to route data key queries to a separate
 MongoDB cluster.
 
+If a ``keyVaultClient`` is not passed, and the parent ``MongoClient`` is
+configured with a non-zero ``maxPoolSize``, the ``keyVaultClient`` is set to an
+internal ``MongoClient``. See `keyVaultClient, metadataClient, and the internal
+MongoClient`_ for configuration behavior.
+
+See `What's the deal with metadataClient, keyVaultClient, and the internal client?`_
+
+keyVaultClient, metadataClient, and the internal MongoClient
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The following pseudo-code describes the configuration behavior for the three ``MongoClient``s:
+
+.. code::
+
+   def getOrCreateInternalClient (client, clientOpts):
+      if client.internalClient != None:
+         return client.internalClient
+      internalClientOpts = copy(clientOpts)
+      internalClientOpts.autoEncryptionOpts = None
+      internalClientOpts.minPoolSize = 0
+      client.internalClient = MongoClient (internalClientOpts)
+      return client.internalClient
+
+   def configureAutoEncryptionClients (client, clientOpts):
+      if clientOpts.autoEncryptionOpts.keyVaultClient != None:
+         client.keyVaultClient = clientOpts.autoEncryptionOpts.keyVaultClient
+      elif clientOpts.maxPoolSize == 0:
+         client.keyVaultClient = client
+      else:
+         client.keyVaultClient = getOrCreateInternalClient (client, clientOpts)
+
+      if clientOpts.autoEncryptionOpts.bypassAutomaticEncryption:
```

Comment:
the new value from AutoEncryptionOpts is not used?

Reply:
I meant this value: https://github.com/mongodb/specifications/pull/888/files#diff-de01d398114f63480af23798e09ffca599aa758efd04f15a9beec89b4566041bR252

Reply:
Ah right! That was leftover from a prior change and should not be in `AutoEncryptionOpts`. It is removed.